BlazarBase/src/main/java/com/hubspot/blazar/base/ModuleState.java

```diff
@@ -46,6 +49,14 @@ public ModuleState(@JsonProperty("module") Module module,
     this.pendingBranchBuild =  filterBranchBuild(pendingBranchBuild);
   }
 
+  /* When this object is built by rosetta the missing build rows are filled with nulls,
+   * which results in objects like this being passed to ModuleState's constructor:
+   *
+   * ModuleBuild(Optional.absent(), 0, 0, 0, null, Optional.absent()...)
+   *
+   * This results in a "present" ModuleBuild, but it has no actual values, the filters here
+   * find such elements and replace them with just Optional.absent()
+   */
```

Comment:
@gchomatas I think the `IllegalArgumentExceptions` in the constructors for `moduleBuild` and `repositoryBuild` are going to break the module state query.